File: src/MyServiceBus/Fault.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace MyServiceBus;

public class Fault<T>
    where T : class
{
    [JsonPropertyName("message")] public T Message { get; set; }

    [JsonPropertyName("exceptions")] public List<ExceptionInfo> Exceptions { get; set; } = new();

    [JsonPropertyName("host")] public HostInfo? Host { get; set; }

    [JsonPropertyName("messageId")] public Guid? MessageId { get; set; }

    [JsonPropertyName("sentTime")] public DateTimeOffset SentTime { get; set; }

    [JsonPropertyName("faultId")] public Guid FaultId { get; set; }

    [JsonPropertyName("conversationId")] public Guid? ConversationId { get; set; }

    [JsonPropertyName("correlationId")] public Guid? CorrelationId { get; set; }
}

public class ExceptionInfo
{
    [JsonPropertyName("exceptionType")] public string ExceptionType { get; set; } = string.Empty;

    [JsonPropertyName("message")] public string Message { get; set; } = string.Empty;

    [JsonPropertyName("stackTrace")] public string? StackTrace { get; set; }

    [JsonPropertyName("innerException")] public ExceptionInfo? InnerException { get; set; }

    public static ExceptionInfo FromException(Exception exception) => new ExceptionInfo
    {
        ExceptionType = exception.GetType().FullName ?? string.Empty,
        Message = exception.Message,
        StackTrace = exception.StackTrace,
        InnerException = exception.InnerException != null ? FromException(exception.InnerException) : null
    };
}

File: src/Java/myservicebus-abstractions/src/main/java/com/myservicebus/Fault.java
package com.myservicebus;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Fault<T> {

    @JsonProperty("message")
    private T message;

    @JsonProperty("exceptions")
    private List<ExceptionInfo> exceptions = new ArrayList<>();

    @JsonProperty("host")
    private HostInfo host;

    @JsonProperty("faultedMessageId")
    private UUID messageId;

    @JsonProperty("timestamp")
    @JsonFormat(shape = JsonFormat.Shape.STRING)
    private OffsetDateTime sentTime;

    @JsonProperty("faultId")
    private UUID faultId;

    @JsonProperty("conversationId")
    private UUID conversationId;

    @JsonProperty("correlationId")
    private UUID correlationId;
}
